webapp/sources/rudder/rudder-web/src/main/elm/onboarding/sources/JsonDecoder.elm

```diff
@@ -0,0 +1,27 @@
+module JsonDecoder exposing (..)
+
+import DataTypes exposing (..)
+import Json.Decode as D exposing (Decoder, andThen, fail, string, succeed)
+import Json.Decode.Pipeline exposing (required, optional)
+import String exposing (toLower)
+
+decodeGetAccountSettings : Decoder AccountSettings
+decodeGetAccountSettings =
+  D.at [ "data", "pluginSettings" ] decodeAccountSettings
+
+decodeAccountSettings : Decoder AccountSettings
+decodeAccountSettings =
+  D.succeed AccountSettings
+    |> required "username"      D.string
+    |> required "password"      D.string
+    |> required "url"           D.string
+    |> optional "proxyUrl"      (D.map Just string) Nothing
+    |> optional "proxyUser"     (D.map Just string) Nothing
+    |> optional "proxyPassword" (D.map Just string) Nothing
+decodeGetMetricsSettings : Decoder String
```

Comment:
You should check that the value is one of the valid values like here: https://github.com/Normation/rudder/pull/3524/files#diff-4904be78e12773cf90b4b3cd05e828141527a0dc55ea6a53581d99b8d67c1142R625

and decode either to a Maybe MetricsState or MetricsState depending if you have 4 or 3 values (see other comment in DataTypess.elm)